backend/src/apiserver/model/pipeline_version.go

```diff
@@ -0,0 +1,65 @@
+package model
+
+import "fmt"
+
+// PipelineVersionStatus a label for the status of the Pipeline.
+// This is intend to make pipeline creation and deletion atomic.
+type PipelineVersionStatus string
+
+const (
+	PipelineVersionCreating PipelineVersionStatus = "CREATING"
+	PipelineVersionReady    PipelineVersionStatus = "READY"
+	PipelineVersionDeleting PipelineVersionStatus = "DELETING"
+)
+
+type PipelineVersion struct {
+	UUID           string `gorm:"column:VersionUUID; not null; primary_key"`
+	CreatedAtInSec int64  `gorm:"column:VersionCreatedAtInSec; not null"`
+	Name           string `gorm:"column:VersionName; not null"`
```

Comment:
maybe we've discussed about this. the name of the version should be unique for a given pipeline for caching purpose. can we enforce it through db schema?

Reply:
yes, we did. I missed it in the change..... added "unique"